packages/tracing/src/integrations/node/express.ts

```diff
@@ -211,3 +239,71 @@ function patchMiddleware(router: Router, method: Method): Router {
 }
+
+/**
+ * Patches the prototype of Express.Router to accumulate the resolved route
+ * if a layer instance's `match` function was called and it returned a successful match.
+ *
+ * @see https://github.com/expressjs/express/blob/master/lib/router/index.js
+ *
+ * @param appOrRouter the router instance which can either be an app (i.e. top-level) or a (nested) router.
+ */
+function instrumentRouter(appOrRouter: ExpressRouter): void {
+  // This is how we can distinguish between app and routers
+  const isApp = 'settings' in appOrRouter;
+
+  // In case the app's top-level router hasn't been initialized yet, we have to do it now
+  if (isApp && appOrRouter._router === undefined) {
+    appOrRouter.lazyrouter && appOrRouter.lazyrouter();
+  }
+
+  const router = isApp ? appOrRouter._router : appOrRouter;
+  const routerProto = Object.getPrototypeOf(router) as ExpressRouter;
+
+  const originalProcessParams = routerProto.process_params;
+  routerProto.process_params = function process_params(
+    layer: Layer,
+    called: unknown,
+    req: PatchedRequest,
+    res: ExpressResponse & SentryTracingResponse,
+    done: () => unknown,
+  ) {
+    // Base case: We're in the first part of the URL (thus we start with the root '/')
+    if (!req._reconstructedRoute) {
+      req._reconstructedRoute = '/';
+    }
+
+    // If the layers partial route has params, it's stored in layer.route, else it's in layer.path
+    const partialRoute = layer.route?.path || layer.path || '';
+
+    // Normalize the partial route so that it doesn't contain leading or trailing slashes
+    // and exclude empty or '*' wildcard routes.
+    // The exclusion of '*' routes is our best effort of not "polluting" the transaction name
+    // with interim handlers (e.g. that check authentication or do other middleware stuff)
+    // We want to end up with the parameterized URL of the incoming request with nothing in between
+    const finalPartialRoute = partialRoute
+      .split('/')
+      .filter(segment => segment.length > 0 && !segment.includes('*'))
+      .join('/');
+
+    // If we found a valid partial URL, we append it to the reconstructed route
+    if (finalPartialRoute.length > 0) {
+      req._reconstructedRoute += `${finalPartialRoute}/`;
+    }
+
+    // Now we check if we are in the "last" part of the route. We determine this by comparing the
+    // numbers of URL segments from the original URL against our reconstructed parameterized URL.
+    // In case we've reached our final destination, we update the transaction name.
+    const urlLength = req.originalUrl?.split('/').filter(s => s.length > 0).length;
+    const routeLength = req._reconstructedRoute.split('/').filter(s => s.length > 0).length;
+    if (urlLength === routeLength) {
+      const transaction = res.__sentry_transaction;
+      if (transaction && transaction.metadata.source !== 'custom') {
+        const finalRoute = req._reconstructedRoute.replace(/\/$/, '');
```

Comment:
Can you add a comment about what you're doing here and why? (I mean, I can figure out the regex, but why is this stripping necessary? Also, easier for the reader if they don't have to translate the regex, however simple.)

Reply:
Thanks for mentioning this. I rewrote the logic slightly and it turns out that we don't need this regex here at all. Will make a small remark on this line nevertheless